### PR TITLE
address technical feedback on API caching blog post

### PR DESCRIPTION
**Pull request by vkarpov15**
## What does this PR do?

Addresses feedback from Ryland and Loren in Slack

See https://github.com/vkarpov15/temporal-api-caching-example/commit/0d8b84c17b17d50c6d7fb16f7796bc69c884899e and https://github.com/vkarpov15/temporal-api-caching-example/commit/439c3316d06c8efe728c8b187714e71efe1dbc22 for corresponding changes in the sample code

### CODE REVIEW

File: blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md [hunk 1]
```diff
@@ -124,6 +127,49 @@ async function run() {
 }
 ```
 
+Following is the output of `start-workflow.ts`.
+
+```
+Started workflow exchange-rates-workflow
+```
+
+Once you have started the Workflow, you can execute a Query to get the latest exchange rates using the following `query-workflow.ts` script.
```
Comment (djmagee) [inline blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md:136]:
```suggestion
After you start the Workflow, you can execute a Query to get the latest exchange rates using the following `query-workflow.ts` script.
```

File: blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md [hunk 2]
```diff
@@ -176,14 +221,18 @@ If the machine running `exchangeRatesWorkflow()` crashes, Temporal can resume th
 
 The `exchangeRatesWorkflow` can run for an unlimited period of time: days, months, even years.
 However, Temporal caps a Workflow at 50,000 events. (See the [Time constraints](https://docs.temporal.io/workflows#time-constraints) section in [Temporal Workflows](https://docs.temporal.io/workflows).)
-In the `exchangeRatesWorkflow`, four events are fired at every iteration of the `while` loop, assuming no API errors.
+In the `exchangeRatesWorkflow`, eight events are fired at every iteration of the `while` loop, assuming no API errors.
```
Comment (djmagee) [inline blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md:224]:
```suggestion
In the `exchangeRatesWorkflow`, eight events are fired during each iteration of the `while` loop, assuming no API errors.
```

File: blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md [hunk 3]
```diff
@@ -176,14 +221,18 @@ If the machine running `exchangeRatesWorkflow()` crashes, Temporal can resume th
...
-With one API request per day, the `exchangeRatesWorkflow()` can run for almost 12,500 days (approximately 34 years) before running into the 50,000 event limit.
+2. `EVENT_TYPE_ACTIVITY_TASK_SCHEDULED`: the Temporal server scheduled the `getExchangeRates()` activity
+3. `EVENT_TYPE_ACTIVITY_TASK_STARTED`: the `getExchangeRates()` activity started executing
+4. `EVENT_TYPE_ACTIVITY_TASK_COMPLETED`: the `getExchangeRates()` activity completed successfully
+5. `EVENT_TYPE_WORKFLOW_TASK_SCHEDULED`: the Temporal server scheduled the Workflow logic that handles `getExchangeRates()`
+6. `EVENT_TYPE_WORKFLOW_TASK_STARTED`: the Workflow resumed
+7. `EVENT_TYPE_WORKFLOW_TASK_COMPLETED`: the Workflow paused
+8. `EVENT_TYPE_TIMER_STARTED`: the Workflow used `setTimeout()` to pause until tomorrow
+
+With one API request per day, the `exchangeRatesWorkflow()` can run for almost 6,250 days (approximately 17 years) before running into the 50,000 event limit.
```
Comment (djmagee) [inline blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md:235]:
```suggestion
With one API request per day, `exchangeRatesWorkflow()` can run for almost 6,250 days (approximately 17 years) before running into the 50,000 event limit.
```

**Review by lorensr — APPROVED** [review]
I'd change this to 1k:

```
const maxIterations = 10000;
```

File: blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md [hunk 4]
```diff
@@ -72,10 +72,12 @@ const { getExchangeRates } = proxyActivities<typeof activities>({
...
 
-export const getExchangeRatesQuery = defineQuery<any, [string]>('getExchangeRates');
+type ExchangeRatesType = { [key: string]: number };
 
-export async function exchangeRatesWorkflow(): Promise<any> {
-  let rates: any = null;
+export const getExchangeRatesQuery = defineQuery<ExchangeRatesType | undefined, [string]>('getExchangeRates');
+
+export async function exchangeRatesWorkflow(): Promise<void> {
+  let rates: ExchangeRatesType | undefined = null;
```
Comment (bergundy) [inline blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md:80]: Did you run this code throw a TS compiler?

```suggestion
  let rates: ExchangeRatesType | undefined = undefined;
```

File: blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md [hunk 5]
```diff
@@ -89,8 +91,7 @@ export async function exchangeRatesWorkflow(): Promise<any> {
     const tomorrow = new Date(today);
     tomorrow.setHours(12, 0, 0, 0);
     tomorrow.setDate(tomorrow.getDate() + 1);
-    // @ts-ignore
-    await sleep(tomorrow - today);
+    await sleep(tomorrow.valueOf() - today.valueOf());
```
Comment (bergundy) [inline blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md:94]: Also could abstract this to some helper function that doesn't need to be implemented for this post

File: blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md [hunk 6]
```diff
@@ -124,6 +127,49 @@ async function run() {
...
+import { WorkflowClient } from '@temporalio/client';
+import { getExchangeRatesQuery } from '../workflows';
+import { toYYYYMMDD } from '../shared';
+
+const date: string = process.argv[2] || toYYYYMMDD(new Date()); 
+
+run().catch((err) => {
+  console.error(err);
+  process.exit(1);
+});
```
Comment (bergundy) [inline blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md:148]: nit: with ESM you can use top level awaits and get rid of this `run` wrapper

**Review by bergundy — APPROVED** [review]
Some minor comments but otherwise LGTM

File: blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md (hunk 3 above)
Comment (flossypurse) [inline blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md:235]: My recent understanding is that in reality we want to keep it under 500.

Comment (lorensr) [reply blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md:235]: Event history length under 500? Due to replay time?

**Review by rylandg — APPROVED** [review]
LGTM

**Pull request merged**